### PR TITLE
Add extended key usage extension to certificate

### PR DESCRIPTION
**Pull request by willyzha**

### CODE REVIEW

File: dpe/src/x509.rs [hunk 1]
```diff
@@ -895,6 +930,51 @@ impl X509CertWriter<'_> {
...
+
+        // Extension data is sequence -> octet string. To compute size, wrap
+        // in tagging twice.
+        bytes_written += self.encode_byte(Self::OCTET_STRING_TAG)?;
+        bytes_written += self.encode_size_field(Self::get_structure_size(
+            Self::get_structure_size(policy_oid.len(), /*tagged=*/ true)?,
+            /*tagged=*/ true,
+        )?)?;
+
+        // Sequence size to just a tagged bool
```
Comment (jhand2) [inline dpe/src/x509.rs:966]: Not sure I understand this comment

File: dpe/src/x509.rs [hunk 2]
```diff
@@ -895,6 +930,51 @@ impl X509CertWriter<'_> {
         Ok(bytes_written)
     }
 
+    /// Encode ExtendedKeyUsage extension
```
Comment (jhand2) [inline dpe/src/x509.rs:933]: Could you update the comment to explain which EKU OID will be included (based on whether cert is_ca)

File: dpe/src/x509.rs [hunk 3]
```diff
@@ -1323,5 +1404,14 @@ mod tests {
             Ok(None) => panic!("key usage extension not found"),
             Err(_) => panic!("multiple key usage extensions found"),
         }
+
+        match cert.extended_key_usage() {
+            Ok(Some(ext_key_usage)) => {
+                assert!(ext_key_usage.critical);
+                assert_eq!(ext_key_usage.value.other, [oid!(2.23.133 .8 .7)]);
```
Comment (jhand2) [inline dpe/src/x509.rs:1411]: nit: Can you add a comment of which OID this is?

**Review by jhand2 — APPROVED** [review]

File: dpe/src/x509.rs (hunk 2 above)
Comment (willyzha) [reply dpe/src/x509.rs:933]: Done

File: dpe/src/x509.rs (hunk 3 above)
Comment (willyzha) [reply dpe/src/x509.rs:1417]: Done.

File: dpe/src/x509.rs (hunk 1 above)
Comment (willyzha) [reply dpe/src/x509.rs]: I think I copied that from somewhere :) updated the comment

**Review by jhand2 — APPROVED** [review]

**Pull request merged**